src/text-input/src/TextInput.js

```diff
@@ -88,25 +116,62 @@ class TextInput extends PureComponent {
     const borderRadius = theme.getBorderRadiusForControlHeight(height)
 
     return (
-      <Text
-        is="input"
-        className={cx(themedClassName, className)}
-        type="text"
-        size={textSize}
-        width={width}
+      <Box
+        display="inline-flex"
+        flex={1}
+        position="relative"
+        width="auto"
         height={height}
-        required={required}
-        disabled={disabled}
-        placeholder={placeholder}
-        paddingLeft={Math.round(height / 3.2)}
-        paddingRight={Math.round(height / 3.2)}
-        borderRadius={borderRadius}
-        spellCheck={spellCheck}
-        aria-invalid={isInvalid}
-        {...(disabled ? { color: 'muted' } : {})}
-        css={css}
-        {...props}
-      />
+      >
+        <Text
+          is="input"
+          className={cx(themedClassName, className)}
+          type="text"
+          size={textSize}
+          width={width}
+          height={height}
+          required={required}
+          disabled={disabled}
+          placeholder={placeholder}
+          paddingLeft={Math.round(height / 3.2)}
+          paddingRight={Math.round(height / 3.2)}
+          borderRadius={borderRadius}
+          spellCheck={spellCheck}
+          aria-invalid={isInvalid}
+          {...(disabled ? { color: 'muted' } : {})}
+          css={css}
+          {...props}
+          value={this.handleValueCase()}
```

Comment:
@jeroenransijn  This is kind of a style choice I would love feedback on here with `handleValueCase()` .. I think a handler like this might make sense, but I did not do the same with `onChange`

Reply:
I will need to spend some more time playing around with this, in this case it's semantically not a handler. As in, it doesn't respond to a event (`onChange`, `onKeyDown` etc.). In this case we could rename it to `this.getValue()` or something.